### PR TITLE
[[ SVGMarkers ]] Implement support for SVG line markers and marker scale

### PR DESCRIPTION
**Pull request by montegoulding**

### CODE REVIEW

**Comment by montegoulding** [comment]
Looks like I broke a couple of tests with this. Marking as WIP until I get a chance to look at the issue.

**Comment by montegoulding** [comment]
I'm not sure why these tests failed as I can't get them to fail locally for some reason. So... I'll poke travis

**Comment by montegoulding** [comment]
It turns out the test just needed to load the SVG path library

**Comment by peter-b** [comment]
@montegoulding The extensions' release notes use "version 2" release note fragments.  See the navbar widget's release notes for example.

**Comment by montegoulding** [comment]
Thanks @peter-b I'll check it out

**Comment by montegoulding** [comment]
@peter-b I think you might need to point me to some actual documentation on the "version 2" release note fragments because I can't really see the problem at the moment with my files.

**Comment by peter-b** [comment]
@montegoulding I don't think we wrote any documentation yet. 😞 

The release note fragment consists of a set of titles which are merged with all the other release notes for a particular subsystem.  For example, suppose I have two release note fragments:

``` markdown
# Top Level Title
## Sub title

* Description of change
```

``` markdown
# Top Level Title
## Sub title

* Description of another change
```

These are merged into a single Markdown file:

``` markdown
# Top Level Title
## Sub title

* Description of change

* Description of another change
```

You can add bug report references to a V2 release note fragment using the syntax:

``` markdown
# [<bug id>] <description>
```

For example,

``` markdown
# [18054] Ensure that appearance reflects loaded state
```

In this particular case, you probably want to have a single release note that looks something like:

``` markdown
# Markers

* The new **markerScale** property controls the size of graph point markers.

* Any named icon from the SVG icon library can now be used as a graph point marker. 
```

**Comment by montegoulding** [comment]
OK, thanks @peter-b ... looks like you have the beginnings of some documentation there ;-)

**Comment by peter-b** [comment]
@montegoulding The most important thing is to choose good heading titles so that the merged release notes don't look like a dog's breakfast ;-)

**Comment by montegoulding** [comment]
@peter-b some dogs eat better than others ;-)

**Comment by montegoulding** [comment]
Rebased onto current develop head

File: extensions/widgets/graph/graph.lcb [hunk 1]
```diff
@@ -35,12 +35,13 @@ widget com.livecode.widget.lineGraph
...
 use com.livecode.widget
 use com.livecode.engine
+use com.livecode.library.iconSVG
 use com.livecode.library.widgetutils
 --
 
 -- adding metadata to ensure the extension displays correctly in LiveCode
 metadata author is "LiveCode"
-metadata version is "1.1.0"
+metadata version is "1.3.0"
```
Comment (peter-b) [inline extensions/widgets/graph/graph.lcb]: Should this be 1.2.0?

File: extensions/widgets/graph/graph.lcb [hunk 2]
```diff
@@ -1053,6 +1098,29 @@ private handler drawGraph() returns nothing
...
+				if mMarkerStyles[tGraphLine - 1] contains "diamond" then
+					rotate tMarkerPath by 45
+				end if
+				
+				put mMarkerStyles[tGraphLine - 1] contains "filled" into tFilledMarkers
+			else
+				put path iconSVGPathFromName(mMarkerStyles[tGraphLine - 1]) into tMarkerPath
+				put true into tFilledMarkers
+				constrainPathToRect(rectangle [-tMarkerSize,-tMarkerSize,tMarkerSize,tMarkerSize],tMarkerPath)
+			end if
```
Comment (peter-b) [inline extensions/widgets/graph/graph.lcb:1124]: I think this code might be easier to read if you put `mMarkerStyles[tGraphLine - 1]` into a local variable within the `if mVerticesShow is true then...` block.

**Comment by peter-b** [comment]
@montegoulding Please resolve conflicts in this PR.

**Comment by peter-b** [comment]
@livecode-vulcan review ok 2c42114

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 2c42114b128aff318e50c1d903133cdb47dd197c

**Comment by livecode-vulcan** [comment]
😎 test success 2c42114b128aff318e50c1d903133cdb47dd197c
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/448)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/453)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/448)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/448)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/454)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/448)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/448)

**Pull request merged**